src/pages/Customer.js

```diff
@@ -143,28 +147,46 @@ class CustomerPage extends Component {
   }
 
   handleDocumentNumber = (event) => {
-    const { documentLabel } = this.state
+    const { key } = event
     let document = 'CPF'
-    let value = path(['target', 'value'], event)
+    let value = clean(path(['target', 'value'], event))
+    const documentLength = value.length
+    const hasDocumentReachedLengthLimit = documentLength >= 11
+
+    const pressedNumber = item => item.match(/[0-9]/)
+    const pressedBackspace = item => item === 'Backspace'
+    const pressedDelete = item => item === 'Delete'
+
+    const shouldUpdate = anyPass([
+      pressedNumber,
+      pressedBackspace,
+      pressedDelete,
+    ])(key)
+
+    if (!shouldUpdate) {
+      return
+    }
 
-    if (documentLabel === 'CPF/CNPJ') {
-      const hasCPFLength = clean(value).length >= 11
+    if (hasDocumentReachedLengthLimit) {
+      this.changePress += 1
+    } else {
+      this.changePress = 0
+    }
 
-      if (hasCPFLength) {
-        this.changePress += 1
-      } else {
-        this.changePress = 0
-      }
+    if (this.changePress > 1) {
+      document = 'CNPJ'
 
-      if (hasCPFLength && this.changePress > 1) {
-        document = 'CNPJ'
-        value = `${value}${event.key}`
+      if (documentLength === 11) {
```

Comment:
@allangrds o que acha de colocar isso em um único `if`?

```suggestion
      if (this.changePress > 1 && documentLength === 11) {
```